src/node.js

```diff
@@ -20,6 +20,32 @@
     });
     EventEmitter.call(process);
 
+    // Default process warning handler
+    process.on('warn', (warning) => {
+      const cluster = NativeModule.require('cluster');
+      if (cluster.isWorker) {
+        // forward warnings on to the controlling process
+        const msg = {
+          cmd: 'NODE_WARN',
+          name: warning.name,
+          message: warning.message,
+          stack: warning.stack
+        };
+        process.send(msg);
+      } else {
+        if (!process.noProcessWarnings) {
+          var message = warning.message;
+          if ((warning.name === 'DeprecationWarning' &&
+              !process.traceDeprecation) ||
+              !process.traceProcessWarnings) {
+            console.error(`${warning.name}: ${message}`);
+          } else {
+            console.error(warning.stack);
+          }
+        }
+      }
+    });
```

Comment:
I still think this should _probably_ only print if `stderr.isTTY`. Might mean you need to mock that for some tests.